src/execution_plan/execution_plan.c

```diff
@@ -91,9 +91,24 @@ static ExecutionPlan *_ExecutionPlan_UnionPlans(AST *ast) {
 	const cypher_astnode_t *union_clause = AST_GetClause(ast, CYPHER_AST_UNION,
 														 NULL);
 	if(!cypher_ast_union_has_all(union_clause)) {
-		OpBase *distinct_op = NewDistinctOp(plan);
-		ExecutionPlan_AddOp(results_op, distinct_op);
-		parent = distinct_op;
+		uint clause_count = cypher_ast_query_nclauses(ast->root);
+		const cypher_astnode_t *last_clause = cypher_ast_query_get_clause(ast->root, clause_count - 1);
+		if(cypher_astnode_type(last_clause) == CYPHER_AST_RETURN) {
```

Comment:
Shouldn't this change to: `ASSERT(cypher_astnode_type(last_clause) == CYPHER_AST_RETURN)` ?

Reply:
It's valid Cypher to write a query like:
```
CREATE (:L)
UNION
CREATE (:M)
```
Though for RedisGraph this currently causes the AST validation failure:
`Found 1 UNION clauses but only 0 RETURN clauses.`

While this doesn't seem like a terribly useful construction, I can see it perhaps being used to guarantee the atomicity of multiple subqueries, so it should probably be supported.